src/ui/SmartSnippet/SmartSnippet.ts

```diff
@@ -98,7 +102,44 @@ export class SmartSnippet extends Component {
      * Any part of an answer exceeding this height will be hidden by default and expendable via a "show more" button.
      * Default value is `250`.
      */
-    maximumSnippetHeight: ComponentOptions.buildNumberOption({ defaultValue: 250, min: 0 })
+    maximumSnippetHeight: ComponentOptions.buildNumberOption({ defaultValue: 250, min: 0 }),
+
+    /**
+     * The field to display for the title.
+     */
+    titleField: ComponentOptions.buildFieldOption({ defaultValue: '@title' }),
+
+    /**
+     * The field to display and use for the URI.
+     */
+    uriField: ComponentOptions.buildFieldOption({ defaultValue: '@clickUri' }),
```

Comment:
I think we can remove uriField and just keep hrefTemplate.

With hrefTemplate, you can do everything uriField allows you to do anyway... It's redundant.

Reply:
I forgot to mention that `uriField` affects the look of the URI as well as the href while `hrefTemplate` only affects the href but has priority. Should `hrefTemplate` also change the visual URI?

Reply:
I agree, it would make more sense to make it so that the uri and hrefTemplate are set using just one param, if possible. Since anyway, you would not want to show one URL and then send the user to a different one.